server/sonar-server/src/main/java/org/sonar/server/db/migrations/AddColumnsBuilder.java

```diff
@@ -0,0 +1,155 @@
+ */
+package org.sonar.server.db.migrations;
+
+import org.sonar.core.persistence.dialect.Dialect;
+import org.sonar.core.persistence.dialect.MsSql;
+import org.sonar.core.persistence.dialect.Oracle;
+import org.sonar.core.persistence.dialect.PostgreSql;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+public class AddColumnsBuilder {
+
+  private final Dialect dialect;
+  private final String tableName;
+  private List<ColumnDef> columnDefs = newArrayList();
+
+  public AddColumnsBuilder(Dialect dialect, String tableName) {
+    this.tableName = tableName;
+    this.dialect = dialect;
+  }
+
+  public AddColumnsBuilder addColumn(ColumnDef columnDef) {
+    columnDefs.add(columnDef);
+    return this;
+  }
+
+  public String build() {
+    StringBuilder sql = new StringBuilder().append("ALTER TABLE ").append(tableName).append(" ");
+    addSql(sql);
+    return sql.toString();
+  }
+
+  private void addSql(StringBuilder sql){
+    switch (dialect.getId()) {
+      case PostgreSql.ID:
+        addColumns(sql, "ADD COLUMN ");
+        break;
+      case MsSql.ID:
+        sql.append("ADD ");
+        addColumns(sql, null);
+        break;
+      default:
+        sql.append("ADD (");
+        addColumns(sql, null);
+        sql.append(")");
+    }
+  }
+
+  private void addColumns(StringBuilder sql, @Nullable String columnPrefix) {
+    for (int i = 0; i < columnDefs.size(); i++) {
+      if (columnPrefix != null) {
+        sql.append(columnPrefix);
+      }
+      addColumn(sql, columnDefs.get(i));
+      if (i < columnDefs.size() - 1) {
+        sql.append(", ");
+      }
+    }
+  }
+
+  private void addColumn(StringBuilder sql, ColumnDef columnDef) {
+    sql.append(columnDef.getName()).append(" ").append(typeToSql(columnDef));
+    Integer limit = columnDef.getLimit();
+    if (limit != null) {
+      sql.append(" (").append(Integer.toString(limit)).append(")");
+    }
+    sql.append(columnDef.isNullable() ? " NULL" : " NOT NULL");
+  }
+
+  private String typeToSql(ColumnDef columnDef) {
+    switch (columnDef.getType()) {
+      case STRING:
+        return "VARCHAR";
+      case BIG_INTEGER:
+      if (dialect.getId().equals(Oracle.ID)) {
+        return "NUMBER (38)";
+      } else {
+        return "BIGINT";
+      }
+      default:
+        throw new IllegalArgumentException("Unsupported type : " + columnDef.getType());
+    }
+  }
+
+  public static class ColumnDef {
+    private String name;
+    private Type type;
+    private boolean isNullable;
+    private Integer limit;
+
+    public enum Type {
+      STRING, BIG_INTEGER
+    }
+
+    public ColumnDef setNullable(boolean isNullable) {
+      this.isNullable = isNullable;
+      return this;
+    }
+
+    public ColumnDef setLimit(@Nullable Integer limit) {
+      this.limit = limit;
+      return this;
+    }
+
+    public ColumnDef setName(String name) {
```

Comment:
Can you check if name must always be lower-case ? If yes, then check should be done here.